demisto_sdk/commands/find_dependencies/find_dependencies.py

```diff
@@ -999,6 +1032,176 @@ def _collect_widget_dependencies(pack_widgets: list, id_set: dict, verbose: bool
+                                            exclude_ignored_dependencies: bool = True) -> set:
+        """
+        Collects generic types dependencies.
+        Args:
+            pack_generic_types (list): collection of pack generics types data.
+            id_set (dict): id set json.
+            verbose (bool): Whether to log the dependencies to the console.
+            exclude_ignored_dependencies (bool): Determines whether to include unsupported dependencies or not.
+        Returns:
+            set: dependencies data that includes pack id and whether is mandatory or not.
+        """
+        dependencies_packs = set()
+        if verbose:
+            click.secho('### Generic Types', fg='white')
+
+        for generic_type in pack_generic_types:
+            generic_type_data = next(iter(generic_type.values()))
+            generic_type_dependencies = set()
+
+            related_scripts = generic_type_data.get('scripts', [])
+            packs_found_from_scripts = PackDependencies._search_packs_by_items_names(
+                related_scripts, id_set['scripts'], exclude_ignored_dependencies)
+
+            if packs_found_from_scripts:
+                pack_dependencies_data = PackDependencies. \
+                    _label_as_mandatory(packs_found_from_scripts)
+                generic_type_dependencies.update(pack_dependencies_data)
+
+            related_definitions = generic_type_data.get('definitionId')
+            packs_found_from_definitions = PackDependencies._search_packs_by_items_names_or_ids(
+                related_definitions, id_set['GenericDefinitions'], exclude_ignored_dependencies)
+
+            if packs_found_from_definitions:
+                pack_dependencies_data = PackDependencies. \
+                    _label_as_mandatory(packs_found_from_definitions)
+                generic_type_dependencies.update(pack_dependencies_data)
+
+            related_layout = generic_type_data.get('layout')
+            packs_found_from_layout = PackDependencies._search_packs_by_items_names_or_ids(
+                related_layout, id_set['Layouts'], exclude_ignored_dependencies)
+
+            if packs_found_from_definitions:
+                pack_dependencies_data = PackDependencies. \
+                    _label_as_mandatory(packs_found_from_layout)
+                generic_type_dependencies.update(pack_dependencies_data)
+
+            if generic_type_dependencies:
+                # do not trim spaces from the end of the string, they are required for the MD structure.
+                if verbose:
+                    click.secho(
+                        f'{os.path.basename(generic_type_data.get("file_path", ""))} depends on: {generic_type_dependencies}',
+                        fg='white')
+            dependencies_packs.update(generic_type_dependencies)
+
+        return dependencies_packs
+
+    @staticmethod
+    def _collect_generic_fields_dependencies(pack_generic_fields: list, id_set: dict, verbose: bool,
```

Comment:
Im not sure, but aren't all three of these methods are exactly the same, but for different entities?
Maybe think of imitating the way we handled it at `_collect_widget_dependencies`

Reply:
they are all depend on definitions but the rest of the dependencies are different (types -> layouts, fields - >types, modules -> dashboards)